arm-documentdb/2015-04-08/swagger/documentdb.json

```diff
@@ -336,24 +336,18 @@
           }
         ],
         "responses": {
-          "200": {
-            "description": "The access key was regenerated successfully.",
-            "schema": {
-              "$ref": "#/definitions/DatabaseAccountListKeysResult"
-            }
+          "202": {
+            "description": "Accepted. The regenerate key operation will complete asynchronously."
           }
         }
       }
     },
-    "/subscriptions/{subscriptionId}/providers/Microsoft.DocumentDB/databaseAccountNames/{accountName}": {
+    "/providers/Microsoft.DocumentDB/databaseAccountNames/{accountName}": {
```

Comment:
Hi @amarzavery 
My two cents, because I'm not working in a service team. A lot of services, like Storage you mention or a namespace for Service Bus, actually does not check name under a subscription, but globally. You cannot have two storage accounts with the same name, even under two different subscriptions. I guess it's because the data RestAPI is not ARM and is using this storage name as a discriminator. So I'm not shocked if under the hood we don't use the subscription.
However, we might want to follow convention and rename the method "CheckNameAvailability". The return type will be different of Storage or CDN, but at least the method call will be similar.
My really two cents,